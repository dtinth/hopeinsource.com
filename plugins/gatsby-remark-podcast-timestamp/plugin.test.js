const Remark = require(`remark`);
const toString = require(`mdast-util-to-string`);
const visit = require(`unist-util-visit`);
const plugin = require(`./plugin`);

const remark = new Remark().data(`settings`, {
  commonmark: true,
  footnotes: true,
  pedantic: true,
});

function filterLoc(obj) {
  if (obj.position) delete obj.position;

  Object.keys(obj).forEach((key) => {
    if (typeof obj[key] === "object") filterLoc(obj[key]);
  });

  return obj;
}

describe(`gatsby-remark-podcast-timestamp`, () => {
  it(`doesn't affect normal paragraphs or use of link references`, () => {
    const markdownAST = remark.parse(`hi [asdf]`);
    const transformed = plugin({ markdownAST }, {});
    visit(transformed, `paragraph`, (node) => {
      expect(node).toMatchSnapshot();
    });
  });

  it(`adds id to a markdown header`, () => {
    const markdownAST = remark.parse(`
[00:00] **Henry**: Hi.
`);
    const transformed = plugin({ markdownAST }, {});

    visit(transformed, `paragraph`, (node) => {
      expect(filterLoc(node)).toMatchSnapshot();
    });
  });
});
