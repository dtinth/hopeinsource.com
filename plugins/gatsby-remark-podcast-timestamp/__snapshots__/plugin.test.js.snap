// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gatsby-remark-podcast-timestamp adds id to a markdown header 1`] = `
Object {
  "children": Array [
    Object {
      "children": undefined,
      "type": "html",
      "value": "<a href=\\"#t=00:00\\"><span class=\\"timestamp\\">00:00 üïê</span></a>",
    },
    Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "type": "text",
              "value": "Henry",
            },
          ],
          "type": "strong",
        },
        Object {
          "type": "text",
          "value": ": Hi.",
        },
      ],
      "data": Object {
        "hName": "span",
        "hProperties": Object {
          "className": Array [
            "text",
          ],
        },
      },
      "type": "paragraph",
    },
  ],
  "data": Object {
    "hProperties": Object {
      "className": Array [
        "wrap",
        "right",
      ],
    },
  },
  "type": "paragraph",
}
`;

exports[`gatsby-remark-podcast-timestamp adds id to a markdown header 2`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "type": "text",
          "value": "Henry",
        },
      ],
      "type": "strong",
    },
    Object {
      "type": "text",
      "value": ": Hi.",
    },
  ],
  "data": Object {
    "hName": "span",
    "hProperties": Object {
      "className": Array [
        "text",
      ],
    },
  },
  "type": "paragraph",
}
`;

exports[`gatsby-remark-podcast-timestamp doesn't affect normal paragraphs or use of link references 1`] = `
Object {
  "children": Array [
    Object {
      "position": Position {
        "end": Object {
          "column": 4,
          "line": 1,
          "offset": 3,
        },
        "indent": Array [],
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "type": "text",
      "value": "hi ",
    },
    Object {
      "children": Array [
        Object {
          "position": Position {
            "end": Object {
              "column": 9,
              "line": 1,
              "offset": 8,
            },
            "indent": Array [],
            "start": Object {
              "column": 5,
              "line": 1,
              "offset": 4,
            },
          },
          "type": "text",
          "value": "asdf",
        },
      ],
      "identifier": "asdf",
      "label": "asdf",
      "position": Position {
        "end": Object {
          "column": 10,
          "line": 1,
          "offset": 9,
        },
        "indent": Array [],
        "start": Object {
          "column": 4,
          "line": 1,
          "offset": 3,
        },
      },
      "referenceType": "shortcut",
      "type": "linkReference",
    },
  ],
  "position": Position {
    "end": Object {
      "column": 10,
      "line": 1,
      "offset": 9,
    },
    "indent": Array [],
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "paragraph",
}
`;
